src/Common/tests/System/Net/Http/Http2LoopbackServer.cs

```diff
@@ -115,8 +115,17 @@ public async Task<Frame> ReadFrameAsync(TimeSpan timeout)
 
             // First read the frame headers, which should tell us how long the rest of the frame is.
             byte[] headerBytes = new byte[Frame.FrameHeaderLength];
-            if (!await FillBufferAsync(headerBytes, timeoutCts.Token).ConfigureAwait(false))
+
+            try
+            {
+                if (!await FillBufferAsync(headerBytes, timeoutCts.Token).ConfigureAwait(false))
+                {
+                    return null;
+                }
+            }
+            catch (IOException)
```

Comment:
Seems like it's better to expect specific tests to eat the connection errors if they are expected, rather than just always eat them...

Reply:
I don't necessarily agree with that.  I think it's important that we have a stable "server".  We are testing the client.  There are many cases the server could get an IOException especially if someone starts a test run then interrupts it.  So, I'm fine to have our test servers (including this loopback server) be robust.  Forcing doing this in any test to makes the tests cumbersome.  They are already cumbersome because we have to start a loopback server in each test.

Reply:
There are tests (like DataFrame_NoStream_ConnectionError) with 
```c#
Assert.Null(await server.ReadFrameAsync(TimeSpan.FromSeconds(30)));
```
and they were failing intermittently.